### PR TITLE
add platform.equality/0.0.1

### PR DESCRIPTION
**Pull request by uselessgoddess**
**platform.equality/0.0.1**

Hello! It's a LinksPlatform again and this our new library for publication
[platform.equality](https://github.com/linksplatform/Equality) is one of the libraries of the [LinksPlatform](https://github.com/linksplatform) modular framework, which uses  innovations from the C++20 standard, for use together with the [platform.hashing](https://github.com/linksplatform/Hashing)[in progress] to create dictionary with `std::any` as key and other features which are needed for [platform.collections](https://github.com/linksplatform/Collections)

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by uselessgoddess** [comment]
Something strange in "changed files". Please take a look

**Comment by prince-chrismc** [comment]
Please sync your branch with Conan's master (there's a button on github from your fork page)

File: recipes/platform.equality/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+required_conan_version = ">=1.33.0"
+
+
+class PlatformInterfacesConan(ConanFile):
+    name = "platform.equality"
+    license = "LGPL-3.0-or-later"
+    homepage = "https://github.com/linksplatform/Equality"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = """platform.delegates is one of the libraries of the LinksPlatform modular framework,
+    which uses innovations from the C++20 standard, for slow parody any typing dictionary and others."""
```
Comment (madebr) [inline recipes/platform.equality/all/conanfile.py:15]: Using `"""` will create an awkward newline and white space in the text.
```suggestion
    description = "platform.delegates is one of the libraries of the LinksPlatform modular framework, "
        "which uses innovations from the C++20 standard, for slow parody any typing dictionary and others."
```

File: recipes/platform.equality/all/test_package/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,10 @@
+cmake_minimum_required(VERSION 3.8)
+project(test_package)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup(TARGETS)
+
+add_executable(${PROJECT_NAME} test_package.cpp)
+target_link_libraries(${PROJECT_NAME} CONAN_PKG::gtest)
+target_link_libraries(${PROJECT_NAME} CONAN_PKG::platform.equality)
```
Comment (madebr) [inline recipes/platform.equality/all/test_package/CMakeLists.txt:9]: We tend to not use targets in test packages for uniformity. 
Unless we're testing cmake_find_package modules, which we are not in this case.
```suggestion
conan_basic_setup()

add_executable(${PROJECT_NAME} test_package.cpp)
target_link_libraries(${PROJECT_NAME} ${CONAN_LIBS})
```

File: recipes/platform.equality/all/test_package/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    def requirements(self):
+        self.requires("gtest/cci.20210126")
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def test(self):
+        if not tools.cross_building(self.settings):
```
Comment (madebr) [inline recipes/platform.equality/all/test_package/conanfile.py:18]:
```suggestion
        if not tools.cross_building(self):
```

File: recipes/platform.equality/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
+import os
+
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
```
Comment (madebr) [inline recipes/platform.equality/all/conanfile.py:4]:
```suggestion
from conans import ConanFile, CMake, tools
from conans.errors import ConanInvalidConfiguration
import os
```

File: recipes/platform.equality/all/test_package/CMakeLists.txt (hunk 2 above)
Comment (uselessgoddess) [reply recipes/platform.equality/all/test_package/CMakeLists.txt:9]: Ok, thanks

File: recipes/platform.equality/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+        return 20
+
+    def validate(self):
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler))
+
+        if not minimum_version:
+            self.output.warn("{} recipe lacks information about the {} compiler support.".format(
+                self.name, self.settings.compiler))
+
+        if tools.Version(self.settings.compiler.version) < minimum_version:
```
Comment (madebr) [inline recipes/platform.equality/all/conanfile.py:47]: Won't this fail when `minimum_version` is `None`?

Comment (uselessgoddess) [reply recipes/platform.equality/all/conanfile.py:47]: So can I write it?
```suggestion
        elif tools.Version(self.settings.compiler.version) < minimum_version:
```

Comment (madebr) [reply recipes/platform.equality/all/conanfile.py:47]: I think that will work

**Review by madebr — APPROVED** [review]

File: recipes/platform.equality/all/test_package/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,20 @@
+from conans import ConanFile, CMake, tools
+import os
+
+
+class TestPackageConan(ConanFile):
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake"
+
+    def requirements(self):
+        self.requires("gtest/cci.20210126")
```
Comment (prince-chrismc) [inline recipes/platform.equality/all/test_package/conanfile.py:10]: Please reduce the test_package we lean against writing test code for the library.
Include headers and make an object to test it links 

This is a little on the complicated side

**Review by madebr — APPROVED** [review]

**Review by Konard — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**